### DISCUSSION TITLE
calls recording save in server side?

### DISCUSSION THREAD

**Discussion by solucionesuno** [InnovateAsterisk/Browser-Phone#400] [Q&A]
is any way to save the call recoriding in thye server side?

**Comment by InnovateAsterisk** [comment]
This involves getting under the hood of the call detail records. 

There are two types, local storage (synchronous), and noSql storage (asynchronous). The basic information of each log of data (in what’s called the stream) is saved to the local storage. It’s designed to be quickly retrieved, and is default limited to 50 display records (even if there are more stored). The call recordings are saved to the noSql storage, and are only mapped back to the CDR, meaning without the cdr you would have difficulty finding the call recordings. 

Either way, the code will write the data as needed at different times. You will probably need to tap into a web hook to fire off an event that could retrieve this data, and then use either your own api to save it to your own database, or even just fetch() with a  lambda function.
